### ISSUE TITLE
Status "Completed" in Civi, but iATS server wasn't reached

### ISSUE THREAD

**Issue by highfalutin** [iATSPayments/com.iatspayments.civicrm#204]
CiviCRM 4.7.18, com.iatspayments.civicrm 1.5.3

A staff person used the Civi iATS back office UI to enter two new recurring contributions. Here are the results, pulled from Civi contact, contribution and iATS request/response log tables, and redacted to protect privacy:

invoice _num | display _name | contribution _page_id | contribution _recur_id | ip | cc | customer _code | total | request_datetime | status _label | auth _result | remote _id | response _datetime
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
69b7.... | Person 1 | NULL | 81 | ... | 1234 | | 35.00 | 2017-05-15 09:28:22 | **Completed** | **NULL** | NULL | NULL
e44f.... | Person 2 | NULL | 82 | ... (same as above) | 5678 | | 20.00 | 2017-05-15 09:30:49 | **Completed** | **NULL** | NULL | NULL

In other words, these contributions were recorded as "Completed" in CiviCRM, even though the iATS server was never successfully reached.

We noticed this only after we had upgraded to 4.7.19/1.6.0, and we haven't been able to reproduce the error, but @KarinG suggested I file an issue because it is a fairly big problem in terms of data integrity.

**Comment by adixon** [comment]
Thanks for the report. My initial suspicion is that:
1. iATS did not respond to the transaction request, presumably it didn't even reach the iATS servers.
2. The failure handling code in the CiviCRM/iATS extension incorrectly handled the payment's pending status and converted it to "complete".

I suspect 2. is not going to happen with the new code, and there's also some new verification stuff going on now to compare against the iATS payments records (i.e. some new transaction auditing options).

For now - the best advice I have is to convert those two transactions to failed and review with whoever entered those transactions to see if there were any other clues.

**Comment by KarinG** [comment]
Hold on - just realizing - you were using 1.5.3 on 4.7.x? You need to use 1.6.0 on 4.7.x [check the .xml files for version compatibility]. So the fact that you've not been able to reproduce the error after you did upgrade to 1.6.0 - tells me that this could very well be a non-issue now.
